R/fitNullModel_internal.R

```diff
@@ -89,8 +89,12 @@
     }
 
     out.class <- class(out)
-    nullprep <- nullModelTestPrep(out)
-    out <- c(out, nullprep)
+    if (!return.small) {
+        nullprep <- nullModelTestPrep(out)
+        out <- c(out, nullprep)
+    } else {
+        out <- smallNullModel(out)
+    }
```

Comment:
I think we still need a couple of things that `nullModelTestPrep` does, even in the small case. Specifically, we need to calculate `resid` and `RSS0` (and maybe `Ytilde`).

Speaking of `Ytilde`, I do want to rename this to `resid.cholesky` at some point (to match `resid.marginal` and `resid.conditional`). Not sure if this would be a good time to do so.

Reply:
For renaming `Ytilde`: could be done in the general null model output updates, since the structure will completely change.

Reply:
Where do `resid` and `RSS0` get used when the other components are not? I couldn't find anywhere, but I might be missing something.

Reply:
Is it a problem that running `nullModelSmall` on an existing null model preserves `resid`, `RSS0`, and `Ytilde`, but running `fitNullModel(return.small=TRUE)` returns a null model without them?

Reply:
From a user perspective, I think it would be confusing if there were different sets of list elements in the returned null model from `nullModelSmall` and `fitNullModel(return.small=TRUE)`. That said, the null model report in the analysis pipeline needs `Ytilde` and `resid`.

Reply:
In that case, maybe we should just always run `nullModelTestPrep`, and only remove the NxN matrices?

Reply:
I realize now that the use case I was thinking of is a future use case. Once I implement the "fast variance approximation for score tests" that SAIGE uses, then you would use a small null model to do assocTestSingle (maybe also aggregate?), but it would require the `resid` and `RSS0` elements. 

It may make the most sense to run `nullModelTestPrep` and then remove the NxN matrices. Using `return.small = TRUE` wouldn't save any time, but it would save memory in the resulting object.

Another possibility would be to make `nullModelTestPrep` a separate step / function to be called on a small null model to make it "full", but that would be a bigger change for users and likely doesn't make as much sense.

Reply:
> For renaming `Ytilde`: could be done in the general null model output updates, since the structure will completely change.

Fine with me to wait til we do this.